icd/api/vk_descriptor_set_layout.cpp

```diff
@@ -204,7 +204,8 @@ uint32_t DescriptorSetLayout::GetDynamicBufferDescDwSize(const Device* pDevice)
 {
     // Currently we store the whole buffer SRD in the dynamic section (i.e. user data registers).
     uint32_t size = 0;
-    if (pDevice->GetEnabledFeatures().robustBufferAccess == false)
+    if ((pDevice->GetRuntimeSettings().enableRelocatableShaders == false) &&
+        (pDevice->GetEnabledFeatures().robustBufferAccess == false))
```

Comment:
Can you create a helper function in the device called UseCompactDynamicDescriptors() and use that here and in similar places in this change?

Reply:
Done.